parse_test.go

```diff
@@ -36,18 +36,19 @@ func TestParseMetadata(t *testing.T) {
 			}
 			err = yaml.Unmarshal(data, &metadata)
 			assert.Nil(t, err)
-			if strings.Contains(string(data), "links:") {
-				assert.Greater(t, len(metadata.Links), 0)
-				linkMap := make(map[string]string)
-				for _, link := range metadata.Links {
-					assert.Greater(t, len(link.Results), 0)
-					assert.Greater(t, len(link.URL), 0)
-					checkDuplicationAcrossLinks(t, link, linkMap)
-					resultSet := mapset.NewSet()
-					for _, result := range link.Results {
-						assert.Greater(t, len(result.TestPath), 0)
-						checkDuplicationWithinResults(t, result, resultSet)
-					}
+
+			linkCount := strings.Count(string(data), "links:")
+			assert.Equal(t, linkCount, 1, "YML file should only contain one `links:` key")
```

Comment:
Is it legal to not contain a `links` key? (Previously looks like that was allowed).

If so, need to change this to <= 1. If not, can you change the assert message to "YML file should contain exactly one `links:` key"?

Reply:
We can enforce the `links:` to be exactly one for now. Once we add a new type of metadata, we can change it to <= 1